execute/tokendata/processor.go

```diff
@@ -0,0 +1,64 @@
+package tokendata
+
+import (
+	"context"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+)
+
+type TokenDataObserver interface {
+	Observe(
+		ctx context.Context,
+		observations exectypes.MessageObservations,
+	) (exectypes.TokenDataObservations, error)
+}
+
+// CompositeTokenDataObserver is a TokenDataObserver that combines multiple TokenDataObserver behind the same interface.
+// Goal of that is to support multiple token observers supporting different tokens (e.g. CCTP, MyFancyToken etc)
+type CompositeTokenDataObserver struct {
+	Observers []TokenDataObserver
+}
+
+func (t *CompositeTokenDataObserver) Observe(
+	ctx context.Context,
+	messages exectypes.MessageObservations,
+) (exectypes.TokenDataObservations, error) {
+	// Dummy implementation, don't focus on that
```

Comment:
Probably you want to remove that comment before merging

Reply:
I am intentionally, splitting that into smaller PRs. This one focuses on high-level interfaces and code organization. Followup one with implementation will cover these